spec/conman_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'conman'
+
+describe Conman do
+
+  let (:input)   {StringIO.new("name\naddress\n123456\nemail@mail.com\nnotes\n")}
+  let (:output)  {StringIO.new}
+  let (:conman)  {Conman.new(UI.new(Console.new(input, output)))}
+
+  it "saves the name introduced by the user" do
+    conman.add_contact
+    expect_field(:name, "name")
+  end
+
+  it "saves the address introduced by the user" do
+    conman.add_contact
+    expect_field(:address, "address")
+  end
+
+  it "saves the phone introduced by the user" do
+    conman.add_contact
+    expect_field(:phone, "123456")
+  end
+
+  it "saves the email introduced by the user" do
+    conman.add_contact
+    expect_field(:email, "email@mail.com")
+  end
+
+  it "saves the notes introduced by the user" do
+    conman.add_contact
+    expect_field(:notes, "notes")
+  end
+
+  it "adds two contacts through the main loop" do
+    input.string << "y\n"
```

Comment:
This ties the high-level class `Conman` to the very concrete detail of string input.
Can you think of a better way to abstract away from these details?